entity_extractor.py

```diff
@@ -0,0 +1,72 @@
+"""NER Entity Token Classifier"""
+from typing import Dict, List, Any
+from pydantic import Field
+from transformers import (
+  pipeline,
+  AutoTokenizer,
+  AutoModelForTokenClassification)
+
+class EntityExtractor():
+  """Entity Extractor"""
+
+  _input_text: str = Field(default=None)
+  _predictions: List[Dict[str, Any]] = Field(default=[])
+  _locations: List[str] = Field(default=[])
+  _organsiations: List[str] = Field(default=[])
+  _persons: List[str] = Field(default=[])
+
+  def __init__(
+    self,
+    model_name:str="Jean-Baptiste/roberta-large-ner-english"):
+
+
+    self._model_name = model_name
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    model = AutoModelForTokenClassification.from_pretrained(model_name)
+
+    self._model = pipeline('ner', model=model, tokenizer=tokenizer, aggregation_strategy="simple")
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `EntityExtractor` class is initialized with a model name, but there's no error handling for the case where the model name provided does not correspond to a valid pre-trained model. This could lead to runtime errors if an invalid model name is passed. Consider adding error handling to ensure that the model and tokenizer are successfully loaded.

```diff
+    try:
    tokenizer = AutoTokenizer.from_pretrained(model_name)

    model = AutoModelForTokenClassification.from_pretrained(model_name)
+    except Exception as e:
+        raise ValueError(f"Failed to load model and tokenizer with name {model_name}. Error: {str(e)}")
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->